Userland/Utilities/less.cpp

```diff
@@ -216,8 +227,14 @@ class Pager {
         }
     }
 
+    size_t count_digits_in_number(size_t line_numbers)
+    {
+        return line_numbers > 0 ? static_cast<size_t>(log10(line_numbers)) + 1 : 1;
+    }
+
     size_t write_range(size_t line, size_t subline, size_t length)
```

Comment:
I was able to make the output look at bit more like that of `less` on Linux / FreeBSD, by making the following changes to `write_range`:

* Calculate the column width for each line individually by moving the `digits_count_for_max_line_number` calculation inside the `for` loop. This resolves the issue seen above where numbers  would render incorrectly as they approach 100, although there's definitely more efficient ways to go about this.
* Introduce a minimum column size of 7 to prevent content being pushed to the right as the line number gets larger. I used a minimum width of 7, as this matches the behavior of `less` in Linux and FreeBSD. A lower number could also work here.
* Use the `{:>{}}` format specifier to right-align the numbers.

```c++
    size_t write_range(size_t line, size_t subline, size_t length)
    {
        constexpr size_t min_column_width = 7;
        size_t lines = 0;
        for (size_t i = line; i < m_lines.size(); ++i) {
            auto digits_count_for_max_line_number = count_digits_in_number(i + 1);
            auto column_width = max(min_column_width, digits_count_for_max_line_number);
        
            for (auto string : sublines(i)) {
                if (subline > 0) {
                    --subline;
                    continue;
                }
                if (lines >= length)
                    return lines;

                if (m_show_line_numbers) {
                    out(m_tty, start_bold);
                    out(m_tty, "{:>{}}", i + 1, column_width);
...
```